cldfbench_steinertthrelkeldmodals.py

```diff
@@ -2,44 +2,126 @@
 import itertools
 
 from cldfbench import Dataset as BaseDataset, CLDFSpec
+from pycldf import term_uri
 
 
 class Dataset(BaseDataset):
     dir = pathlib.Path(__file__).parent
     id = "steinertthrelkeldmodals"
 
     def cldf_specs(self):  # A dataset must declare all CLDF sets it creates.
-        return CLDFSpec(module='StructureDataset', dir=self.cldf_dir)
+        return CLDFSpec(module="StructureDataset", dir=self.cldf_dir)
 
     def cmd_makecldf(self, args):
-        args.writer.cldf.add_component('ParameterTable')
-        args.writer.cldf.add_component('LanguageTable')
-        args.writer.cldf.add_columns(
-            'ValueTable',
+        args.writer.cldf.add_component("ParameterTable")
+        args.writer.cldf.add_component("LanguageTable")
+        # new, non-standard tables
+        # NOTE: should the column names use `term_uri` and, if so, how?
+        args.writer.cldf.add_table(
+            "unit-parameters.csv",
             {
-                "name": "expressivity",
-                "datatype": "json",
-            }
+                "name": "ID",
+                "propertyUrl": "http://cldf.clld.org/v1.0/terms.rdf#id",
+            },
+            "Name",
+            "Description",
+            "force",
+            "flavor",
         )
-        vid = 0
-        args.writer.objects['ParameterTable'].append(dict(ID='modal'))
+        args.writer.cldf.add_table(
+            "unit-values.csv",
+            {
+                "name": "ID",
+                "propertyUrl": "http://cldf.clld.org/v1.0/terms.rdf#id",
+            },
+            {
+                "name": "Value",
+                # FIXME: valueReference should be added to the ontology, in analogy to formReference.
+                #"propertyUrl": "http://cldf.clld.org/v1.0/terms.rdf#valueReference",
+            },
+            "UnitParameter_ID",
+            "UnitValue",
+            "Comment",
+            "Source",
+        )
+        args.writer.cldf.add_table(
+            "flavors.csv",
+            {
+                "name": "ID",
+                "propertyUrl": "http://cldf.clld.org/v1.0/terms.rdf#id",
+            },
+            "Name", "Description"
+        )
+        args.writer.cldf.add_table(
+            "forces.csv",
+            {
+                "name": "ID",
+                "propertyUrl": "http://cldf.clld.org/v1.0/terms.rdf#id",
+            },
+            "Name", "Description"
+        )
+        args.writer.cldf.add_foreign_key(
+            "unit-values.csv", "Value", "ValueTable", "ID"
+        )
+        args.writer.cldf.add_foreign_key(
+            "unit-values.csv", "UnitParameter_ID", "unit-parameters.csv", "ID"
+        )
+        args.writer.cldf.add_foreign_key(
+            "unit-parameters.csv", "flavor", "flavors.csv", "Name" 
```

Comment:
@shanest I typically use the primary key - i.e. an `ID` column specified with propertyUrl `...#id` as target for foreign keys, since this column is guaranteed to be unique.

Reply:
Makes sense; thanks!